File: platforms/python/examples/core/pass/Pass.py
from rendercanvas.auto import RenderCanvas, loop

from fragmentcolor import Shader, Pass, Renderer

renderer = Renderer()
canvas = RenderCanvas(size=(100, 100))
target = renderer.create_target(canvas)
shader = Shader.default()

rpass = Pass("First Pass")
rpass.add_shader(shader)

pass2 = Pass("Second Pass")
pass2.add_shader(shader)

# standalone
renderer.render(rpass, target)

# vector of passes rendered in order (any iterable of Pass is renderable)
renderer.render([rpass, pass2], target)

File: platforms/swift/examples/core/pass/Pass.swift

import FragmentColor

let renderer = Renderer()
// iOS: window/canvas provided by CAMetalLayer at runtime
let target = try await renderer.createTextureTarget([800, 600])
let shader = Shader.default()

let pass = Pass("First Pass")
pass.addShader(shader)

let pass2 = Pass("Second Pass")
pass2.addShader(shader)

// standalone
try renderer.render(pass, target)

// vector of passes rendered in order (any iterable of Pass is renderable)
try renderer.render([pass, pass2], target)